### PR TITLE
SYN-4088: Adds max_wait_time and wait_for_nav_timeout for browser tests

### PR DESCRIPTION
**Pull request by sangn-splunk**
Resolves [https://splunk.atlassian.net/browse/SYN-4088](https://splunk.atlassian.net/browse/SYN-4088)

----

### Before the change?

* Assert steps on browser steps did not allow you to configure a wait time (wait for x seconds for an element to appear). And steps with navigation did not allow you to configure how long to wait for a navigation event.

### After the change?

* Adds max_wait_time and wait_for_nav_timeout to browser steps, allowing users to specify wait times. See https://splunk.enterprise.slack.com/docs/T024FQ3UW/F07BZ0A7TK6 for a small demo.

### Pull request checklist 
- [ ] Acceptance Tests have been updated, run (`make testacc`), and pasted in this PR (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

#### Acceptance Test Output

https://github.com/splunk/syntheticsclient/actions/runs/10048580222/job/27772849666?pr=25

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: syntheticsclientv2/common_models.go [hunk 1]
```diff
@@ -77,6 +77,8 @@ type StepsV2 struct {
 	URL                string  `json:"url,omitempty"`
 	Action             string  `json:"action,omitempty"`
 	WaitForNav         bool    `json:"waitForNav"`
+  WaitForNavTimeout  int     `json:"waitForNavTimeout"`
```
Comment (greatestusername-splunk) [inline syntheticsclientv2/common_models.go]: Thank you for adding this!
Mind hitting these with a linter or just fixing the spacing? 

Sort of surprised our pipeline didn't catch this. May need to tweak that step of the workflow to fail when linting isn't correct.

**Review by greatestusername-splunk — APPROVED** [review]
LGTM other than a small linting thing.

File: syntheticsclientv2/common_models.go (hunk 1 above)
Comment (greatestusername-splunk) [reply syntheticsclientv2/common_models.go]: Hah ok so when I pull the PR as a branch this all looks fine. And Sang see's the same. Lint doesn't show any changes. This must be a Github weirdness?

Ok to merge imo and if it ends up being weird...  can fix in a later PR IMO.

Comment (sangn-splunk) [reply syntheticsclientv2/common_models.go]: (discussed over slack)

i tried `golangci-lint run`, but it didn't change any files. 
image

File: syntheticsclientv2/common_models.go [hunk 2]
```diff
@@ -77,6 +77,8 @@ type StepsV2 struct {
 	URL                string  `json:"url,omitempty"`
 	Action             string  `json:"action,omitempty"`
 	WaitForNav         bool    `json:"waitForNav"`
+	WaitForNavTimeout  int     `json:"waitForNavTimeout"`
```
Comment (greatestusername-splunk) [inline syntheticsclientv2/common_models.go:80]: You did it! Nice!

**Pull request merged**